test/Microsoft.AspNet.Mvc.ModelBinding.Test/Metadata/AssociatedMetadataProviderTest.cs

```diff
@@ -248,6 +284,18 @@ private class PropertyModel
             public double MixedAttributes { get; set; }
         }
 
+        [ModelMetadataType(typeof(PropertyModel))]
+        private class PropertyViewModel
+        {
+            [Range(0, 10)]
+            public int LocalAttributes { get; set; }
+
+            [Range(0, 10)]
+            public string MetadataAttributes { get; set; }
+
+            public double MixedAttributes { get; set; }
+        }
```

Comment:
property names here don't match the applied attributes.  expect `LocalAttributes` to have attributes only in `PropertyViewModel`, `MetadataAttributes` only in `PropertyModel`, and `MixedAttributes` in both.  see the MVC 5 class for example:
``` c#
[MetadataType(typeof(Metadata))]
private class PropertyModel
{
    [Required]
    public int LocalAttributes { get; set; }

    public string MetadataAttributes { get; set; }

    [Required]
    public double MixedAttributes { get; set; }

    private class Metadata
    {
        [Range(10, 100)]
        public object MetadataAttributes { get; set; }

        [Range(10, 100)]
        public object MixedAttributes { get; set; }
    }
}
```